src/client/App.tsx

```diff
@@ -34,15 +37,31 @@ export const App: React.ComponentType<StorageState> = ({ session }) => {
           <StorageContext.Provider value={{ session }}>
             <AppTokenRetrieval>
               <Switch>
-                {reactPageRoutes.map(({ path, exact, Component, render }) => (
-                  <Route
-                    key={path}
-                    path={path}
-                    exact={exact}
-                    component={Component}
-                    render={render}
-                  />
-                ))}
+                {routes.map(({ path, clientRouteData }) => {
+                  if (!clientRouteData || !(typeof path === 'string')) {
+                    return null
+                  }
```

Comment:
Why not filter out these routes? `routes.filter().map()`?

Reply:
Good point 👍

Reply:
OK so TypeScript wasn't very happy about that solution which made the code very ugly with a lot of type assertions etc. so I skipped this.

Reply:
For reference, I faced this issue before and I think type predicates are the recommended way to approach filtering. https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates